dropwizard-testing/src/test/java/io/dropwizard/testing/junit/ServerSideExceptionsCollectorTest.java

```diff
@@ -0,0 +1,84 @@
+package io.dropwizard.testing.junit;
+
+import static io.dropwizard.testing.ResourceHelpers.resourceFilePath;
+import static org.junit.Assert.assertThat;
+import static org.hamcrest.CoreMatchers.equalTo;
+
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
```

Comment:
I believe AssertJ's assertions and exception detection is preferred to junit + hamcrest.

Reply:
The server side exception is only attached after the exception is throw and the `@Rule ServerSideExceptionsCollector` has a chance to run, so, can't test `ServerSideExceptionsCollector` behaviour using `catchingThrowable`